DockWin.ahk

```diff
@@ -0,0 +1,207 @@
+;DockWin v0.5 - Save and Restore window positions when docking/undocking (using hotkeys) over all Windows 10 Desktops
+; Paul Troiano, 6/2014
+; Updated by Ashley Dawson 7/2015
+; Updated by Robert Niebsch 10/2016
+;
+; Hotkeys: Capslock + 0 to save; Capslock + 9 to load
+
+class DockWinClass
+{
+
+    SaveAllWindowsFunctionName := "SaveAllWindows"
+    LoadAllWindowsFunctionName := "LoadAllWindows"
+    
+	
+	__new()
+    {  
+        this.WindowMover := new JPGIncWindowMoverClass()
+		this.desktopMapper := new DesktopMapperClass(new VirtualDesktopManagerClass())
+		this.monitorMapper := new MonitorMapperClass()  
+        return this
+    }
+    
+; goes through file
+; title of line in file is matched with open windows
+LoadAllWindows()
+{
+    SetTitleMatchMode, 3		; 2: A window's title can contain WinTitle anywhere inside it to be a match. 
+    SetTitleMatchMode, Fast		;Fast is default
+    DetectHiddenWindows, off	;Off is default
+
+    SetWorkingDir %A_ScriptDir%  ; Ensures a consistent starting directory.
+    CrLf=`r`n
+    FileName:="WinPos.txt"
+        
+    WinGetActiveTitle, SavedActiveWindow
+    ParmVals:="Title Desktop x y height width maximized path"
+    SectionToFind:= this.SectionHeader()
+    SectionFound:= 0
+    
+    FileSelectFile, FileName, 1, WinPos.txt ;ask when overwriting; default name WinPos.txt

```

Comment:
To save the user having to accept popups all the time you could make FileName a class variable and do something like this:
if(fileexists(this.filename)) {
do stuff with the file
} else {
select the file
save the file name in the class variable so future calls just read the file
}

having this in it's own fuction is even better